src/Strategy/ManifestStrategy.php

```diff
@@ -0,0 +1,213 @@
+use Humbug\SelfUpdate\Exception\JsonParsingException;
+use Humbug\SelfUpdate\Updater;
+use Humbug\SelfUpdate\VersionParser;
+use Humbug\SelfUpdate\Exception\RuntimeException;
+
+final class ManifestStrategy implements StrategyInterface
+{
+    /**
+     * @var array
+     */
+    private static $requiredKeys = array('sha1', 'version', 'url');
+
+    /**
+     * @var string
+     */
+    private $manifestUrl;
+
+    /**
+     * @var array
+     */
+    private $manifest;
+
+    /**
+     * @var array
+     */
+    private $availableVersions;
+
+    /**
+     * @var string
+     */
+    private $localVersion;
+
+    /**
+     * @var bool
+     */
+    private $allowMajor = false;
+
+    /**
+     * @var bool
+     */
+    private $allowUnstable = false;
+
+    /**
+     * ManifestStrategy constructor.
+     *
+     * @param string $localVersion  The local version.
+     * @param string $manifestUrl   The URL to a JSON manifest file. The
+     *                              manifest contains an array of objects, each
+     *                              containing a 'version', 'sha1', and 'url'.
+     * @param bool   $allowMajor    Whether to allow updating between major
+     *                              versions.
+     * @param bool   $allowUnstable Whether to allow updating to an unstable
+     *                              version. Ignored if $localVersion is unstable
+     *                              and there are no new stable versions.
+     */
+    public function __construct($localVersion, $manifestUrl, $allowMajor = false, $allowUnstable = false)
+    {
+        $this->localVersion = $localVersion;
+        $this->manifestUrl = $manifestUrl;
+        $this->allowMajor = $allowMajor;
+        $this->allowUnstable = $allowUnstable;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCurrentLocalVersion(Updater $updater)
+    {
+        return $this->localVersion;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function download(Updater $updater)
+    {
+        $version = $this->getCurrentRemoteVersion($updater);
+        if ($version === false) {
+            throw new RuntimeException('No remote versions found');
+        }
+
+        $versionInfo = $this->getAvailableVersions();
+        if (!isset($versionInfo[$version])) {
+            throw new RuntimeException(sprintf('Failed to find manifest item for version %s', $version));
+        }
+
+        $fileContents = file_get_contents($versionInfo[$version]['url']);
+        if ($fileContents === false) {
+            throw new HttpRequestException(sprintf('Failed to download file from URL: %s', $versionInfo[$version]['url']));
+        }
+
+        $tmpFilename = $updater->getTempPharFile();
+        if (file_put_contents($tmpFilename, $fileContents) === false) {
+            throw new RuntimeException(sprintf('Failed to write file: %s', $tmpFilename));
+        }
+
+        $tmpSha = sha1_file($tmpFilename);
```

Comment:
It's on the todo list. I'm thinking about scrapping sha1 altogether since this is a new feature.